components/dashboard/src/settings/Preferences.tsx

```diff
@@ -16,73 +15,60 @@ import { getGitpodService } from "../service/service";
 import { ThemeContext } from "../theme-context";
 import { UserContext } from "../user-context";
 import settingsMenu from "./settings-menu";
+import IDENone from '../icons/IDENone.svg';
+import IDENoneDark from '../icons/IDENoneDark.svg';
 
 type Theme = 'light' | 'dark' | 'system';
 
+const DesktopNoneId = "none";
+const DesktopNone: IDEOption = {
+    "image": "",
+    "logo": IDENone,
+    "orderKey": "-1",
+    "title": "None",
+    "type": "desktop"
+};
+
 export default function Preferences() {
     const { user } = useContext(UserContext);
-    const { setIsDark } = useContext(ThemeContext);
-
-    const [defaultIde, setDefaultIde] = useState<string>(user?.additionalData?.ideSettings?.defaultIde || "");
-    const actuallySetDefaultIde = async (value: string) => {
-        const additionalData = user?.additionalData || {};
-        const settings = additionalData.ideSettings || {};
-        settings.defaultIde = value;
+    const { setIsDark, isDark } = useContext(ThemeContext);
+
+    const updateUserIDEInfo = async (defaultDesktopIde: string, defaultIde: string) => {
+        const useDesktopIde = defaultDesktopIde !== DesktopNoneId;
+        const additionalData = user?.additionalData ?? {};
+        const settings = additionalData.ideSettings ?? {};
+        settings.useDesktopIde = useDesktopIde;
+        settings.defaultIde = defaultIde;
+        settings.defaultDesktopIde = defaultDesktopIde;
         additionalData.ideSettings = settings;
         getGitpodService().server.trackEvent({
             event: "ide_configuration_changed",
             properties: {
                 useDesktopIde,
-                defaultIde: value,
+                defaultIde,
                 defaultDesktopIde: useDesktopIde ? defaultDesktopIde : undefined
             },
-        });
+        }).then().catch(console.error);
         await getGitpodService().server.updateLoggedInUser({ additionalData });
+    }
+
+    const [defaultIde, setDefaultIde] = useState<string>(user?.additionalData?.ideSettings?.defaultIde || "");
+    const actuallySetDefaultIde = async (value: string) => {
+        await updateUserIDEInfo(defaultDesktopIde, value);
         setDefaultIde(value);
     }
 
     const [defaultDesktopIde, setDefaultDesktopIde] = useState<string>(user?.additionalData?.ideSettings?.defaultDesktopIde || "");
```

Comment:
**Bug**: As a brand new User, I see this in my Preferences page (if I don't change anything):

<img width="300" alt="Screenshot 2022-01-20 at 13 04 48" src="https://user-images.githubusercontent.com/599268/150335559-06345372-3041-4c3d-b472-840717c20f21.png">

You can see that `VS Code` is selected as my default Desktop IDE.

However, if I start a workspace, it goes straight to Browser IDE (it doesn't propose to use my preferred Desktop IDE). To make it propose a Desktop IDE, I need to first select `None`, and then `VS Code` again.

Wouldn't it make more sense to show `None` as selected by default, when `useDesktopIde` is false?

```suggestion
    const [defaultDesktopIde, setDefaultDesktopIde] = useState<string>(user?.additionalData?.ideSettings?.useDesktopIde && user?.additionalData?.ideSettings?.defaultDesktopIde || DesktopNoneId);
```

Reply:
Thanks 🙂